### PR TITLE
minor: add another test case to cover join ambiguous check

### PR DESCRIPTION
**Pull request by ygf11**
# Which issue does this PR close?

Closes #.

* As @mingmwang suggests, add another test case which works with using-join to cover ambiguous check. 
* Improve some test cases. 

# Rationale for this change

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/sql/src/planner.rs [hunk 1]
```diff
@@ -6070,7 +6070,23 @@ mod tests {
...
+            INNER JOIN person as p2 
+            using(id)";
+
+        let expected = "Projection: p1.id, p1.age, p2.id\
+            \n  Inner Join: Using p1.id = p2.id\
+            \n    SubqueryAlias: p1\
+            \n      TableScan: person\
+            \n    SubqueryAlias: p2\
+            \n      TableScan: person";
+        quick_test(sql, expected);
```
Comment (ygf11) [inline datafusion/sql/src/planner.rs:6089]: cc @mingmwang

File: datafusion/sql/src/planner.rs [hunk 2]
```diff
@@ -6057,7 +6057,7 @@ mod tests {
     }
 
     #[test]
-    fn test_ambiguous_coulmn_referece_in_join() {
+    fn test_ambiguous_coulmn_referece_in_on_join() {
```
Comment (mingmwang) [inline datafusion/sql/src/planner.rs:6060]: Should be `column references`

File: datafusion/sql/src/planner.rs (hunk 1 above)
Comment (mingmwang) [reply datafusion/sql/src/planner.rs:6089]: Except for typo,  LGTM.
Thanks for adding the test case.

File: datafusion/sql/src/planner.rs [hunk 3]
```diff
@@ -6070,7 +6070,23 @@ mod tests {
         let result = logical_plan(sql);
         assert!(result.is_err());
         let err = result.err().unwrap();
-        assert_eq!(format!("{}", err), expected);
+        assert_eq!(err.to_string(), expected);
+    }
+
+    #[test]
+    fn test_ambiguous_coulmn_referece_with_in_using_join() {
```
Comment (mingmwang) [inline datafusion/sql/src/planner.rs:6077]: column references

**Comment by ygf11** [comment]
Thanks for your review @mingmwang, I will fix it.

**Review by alamb — APPROVED** [review]
Thanks @ygf11  and @mingmwang 

I am merging this PR in to reduce the review queue -- I'll make a minor PR to fix the typos shortly

**Pull request merged**